app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/vault/util/VaultAddItemStateExtensions.kt

```diff
@@ -118,22 +120,72 @@ private fun VaultAddEditState.ViewState.Content.ItemType.toIdentityView(): Ident
 
 @Suppress("MagicNumber")
 private fun VaultAddEditState.ViewState.Content.toPasswordHistory(): List<PasswordHistoryView>? {
+    val loginCipher = type as? VaultAddEditState.ViewState.Content.ItemType.Login
     val oldPassword = common.originalCipher?.login?.password
+    val timestamp = Instant.now()
+
+    val newPasswordHistory = getPasswordHistory(
+        loginCipher,
+        oldPassword,
+        timestamp,
+    )
+
+    val newHiddenFieldHistory = getHiddenFieldHistory(
+        timestamp,
+        common.originalCipher?.fields,
+        common.customFieldData,
+    )
+
+    val passwordHistory = listOf(
+        common.originalCipher?.passwordHistory.orEmpty(),
+        newPasswordHistory,
+        newHiddenFieldHistory,
+    ).flatten()
+
+    return passwordHistory
+        .ifEmpty { null }
+        ?.takeLast(5)
```

Comment:
Why only 5?

Reply:
It's not a bug, it's a feature. 😄 
It was designed that way.